Octokit/Http/SimpleJsonSerializer.cs

```diff
@@ -90,7 +90,6 @@ public override object DeserializeObject(object value, Type type)
                 var jsonValue = value as JsonObject;
                 if (stringValue != null)
                 {
-                    stringValue = stringValue.Replace("-", "");
```

Comment:
The only reservation I had was that this code was being used elsewhere and some enum values depended on this, but then I saw those places were calling `RemoveHyphenAndUnderscore` and saw it had us covered:

```
static string RemoveHyphenAndUnderscore(string stringValue)
{
    // remove '-' from values coming in to be able to enum utf-8
    stringValue = stringValue.Replace("-", "");
    // remove '-' from values coming in to be able to enum EventInfoState names with underscores in them. Like "head_ref_deleted" 
    stringValue = stringValue.Replace("_", "");
    return stringValue;
}
```